src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -112,6 +134,33 @@ public static Tracer register(final Tracer tracer) {
         return previous;
     }
 
+    /**
+     * Updates the global tracer using the specified {@link UpdateFunction}.
+     * <p>
+     * If the update encounters a race condition, the 'losing' update function is re-applied with the 'winning' tracer.
+     * To avoid concurrency 'hotspins', please make sure the {@linkplain UpdateFunction} is reasonably quick.
+     * <p>
+     * If there are concurrent modifications of the globaltracer (e.g. by a race-condition with a call to
+     * {@link #register(Tracer) register}, the update function will be)
```

Comment:
😄 You're right; I just caught that one too!
I'll push a new version using a ReadWriteLock instead of hotspinning on race conditions.
Then the missing comment doesn't apply.